src/big/lib_gmp.cr

```diff
@@ -1,87 +1,108 @@
-@[Link("gmp")]
+{% if flag?(:win32) %}
+  @[Link("mpir")]
+{% else %}
+  @[Link("gmp")]
+{% end %}
 lib LibGMP
   alias Int = LibC::Int
   alias Long = LibC::Long
   alias ULong = LibC::ULong
+
+  # MPIR uses its own `mpir_si` and `mpir_ui` typedefs in places where GMP uses
+  # the LibC types, when the function name has `si` or `ui`; we follow this
+  # distinction
+  {% if flag?(:win32) && flag?(:x86_64) %}
```

Comment:
I take it, this is not specific to the x86_64 architecture, but for 64-bit architectures in general, so we should use `bits64` flag.
```suggestion
  {% if flag?(:win32) && flag?(:bits64) %}
```
Ditto below.